packages/cxl-ui/src/components/cxl-playbook-accordion.js

```diff
@@ -0,0 +1,156 @@
+  }
+
+  /**
+   *
+   * @return {string}
+   */
+  get checkboxesStorageId() {
+    const attr = this.getAttribute('id') || 'default_id';
+    return `${attr}_checkboxes`;
+  }
+
+  constructor() {
+    super();
+    registerStyles('vaadin-accordion-panel', [vaadinAccordionPanelStyles]);
+    registerStyles('vaadin-checkbox', [vaadinCheckboxStyles]);
+  }
+
+  /**
+   * Global styles.
+   */
+  ready() {
+    super.ready();
+    // Define and register a style sheet for the <vaadin-text-field> component
+    registerGlobalStyles(cxlPlaybookAccordionGlobalStyles, {
+      moduleId: 'cxl-playbook-accordion-global',
+    });
+
+    this._updateCheckboxesStates();
+  }
+
+  /**
+   *
+   * @param {HTMLCollection} items
+   * @private
+   */
+  _saveAccordionState(items) {
+    const storageId = this.getAttribute('id');
+
+    // Avoid null key.
+    if (storageId) {
+      const stateItems = [];
+
+      items.forEach((value, key) => {
+        stateItems[key] = items[key].opened;
+      });
+
+      localStorage.setItem(storageId, JSON.stringify(stateItems));
+
+      this._updateCSSAndPanelStateToCheckboxesStates();
+
+      this._dispatchCustomEvent(stateItems);
+
+      this._saveCheckboxesState();
+    }
+  }
+
+  _saveCheckboxesState() {
+    const stateCheckboxes = [];
+    const checkboxes = this.panelSummarySlotCheckboxes;
+
+    checkboxes.forEach((value, key) => {
+      const checkbox = checkboxes[key];
+
+      stateCheckboxes[key] =
+        checkbox.hasAttribute('aria-checked') && checkbox.getAttribute('aria-checked') === 'true';
+    });
+
+    localStorage.setItem(this.checkboxesStorageId, JSON.stringify(stateCheckboxes));
+  }
+
+  _updateCheckboxesStates() {
+    const stateCheckboxes = JSON.parse(localStorage.getItem(this.checkboxesStorageId));
+
+    if (stateCheckboxes === null) {
+      return;
+    }
+
+    const checkboxes = this.panelSummarySlotCheckboxes;
+
+    checkboxes.forEach((item, key) => {
+      const checkbox = checkboxes[key];
+      const isChecked = !!stateCheckboxes[key]; // autocast from null or undefined to boolean
+
+      checkbox.setAttribute('aria-checked', isChecked ? 'true' : 'false');
+      checkbox.checked = isChecked;
+    });
+
+    this._updateCSSAndPanelStateToCheckboxesStates();
+  }
+
+  _updateCSSAndPanelStateToCheckboxesStates() {
+    const panels = this.accordionPanels;
+
+    this.panelSummarySlotCheckboxes.forEach((checkbox, index) => {
+      if (!panels[index])
+        throw new Error(`vaadin-accordion-panel with index ${index} doesn't exist`);
+
+      const isChecked =
+        checkbox.hasAttribute('aria-checked') && checkbox.getAttribute('aria-checked') === 'true';
+
+      const accordionPanel = panels[index];
+      const summaryTop = accordionPanel.querySelector('.summary-top');
+
+      if (isChecked) {
+        summaryTop.classList.add('checked');
+      } else {
+        summaryTop.classList.remove('checked');
+      }
+    });
+  }
+
+  _dispatchCustomEvent(stateItems) {
```

Comment:
Who is an example consumer of this event? I see zero other uses for this in this PR, what am I missing?

Process: if no uses at this time, avoid putting "hypothetical future things" into PRs - it just makes review longer and harder.

Best PR - delete code
Next best PR - 1 line of code
Next best PR - 2 lines of code

etc

Reply:
> Who is an example consumer of this event? I see zero other uses for this in this PR, what am I missing?
> 
This was designed before `localStorage` state container was implemented here in order to give user the possiblity to catch the event and save the item state to the database via `REST API`. If you don't need this, this can be removed of course.

Reply:
> Process: if no uses at this time, avoid putting "hypothetical future things" into PRs - it just makes review longer and harder.
> 
I was actually going the same way that `VaadinAccordion` is implemented using custom events like `items-changed`
https://vaadin.com/components/vaadin-accordion/html-api/elements/Vaadin.AccordionElement

Reply:
> I was actually going the same way that VaadinAccordion is implemented using custom events like items-changed
https://vaadin.com/components/vaadin-accordion/html-api/elements/Vaadin.AccordionElement

Makes sense, but less code === better, to start. We'd add this when we actually find a use case.